src/Microsoft.AspNet.Authentication/AuthenticationHandler.cs

```diff
@@ -61,12 +64,13 @@ internal AuthenticationOptions BaseOptions
 
         public bool Faulted { get; set; }
 
-        protected async Task BaseInitializeAsync([NotNull] AuthenticationOptions options, [NotNull] HttpContext context, [NotNull] ILogger logger)
+        protected async Task BaseInitializeAsync([NotNull] AuthenticationOptions options, [NotNull] HttpContext context, [NotNull] ILogger logger, [NotNull] IUrlEncoder encoder)
         {
             _baseOptions = options;
             Context = context;
             RequestPathBase = Request.PathBase;
             Logger = logger;
+            UrlEncoder = encoder;
```

Comment:
yeah I don't disagree, but actually you already added that service to all of the samples since cookie requires it, so I didn't really have to do anything extra other than update the bearer one I think.  Perhaps what we need is a dumping ground AddSharedAuthentication extension which we can put things like DataProtection/WebEncoders and just call that consistently.  I don't really like how ConfigureCookieAuthentication is secretly adding web encoder today... that is pretty non standard :)